plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/handlers/FordiacQuickFixHandler.java

```diff
@@ -94,11 +96,26 @@ public static IMarker[] getMarkersFromEditorSelection(final IStructuredSelection
 	public static IMarker getMarker(final EditPart editPart) {
 		if (FordiacAnnotationUtil
 				.getAnnotationModel(editPart) instanceof final FordiacMarkerGraphicalAnnotationModel annotationModel) {
-			final var annotation = annotationModel.getAnnotations(editPart.getModel());
-			if (!annotation.isEmpty()
-					&& annotation.iterator().next() instanceof final GraphicalMarkerAnnotation markerAnnotation) {
-				return markerAnnotation.getMarker();
+
+			final IMarker marker = getMarkerFromAnnotationModel(annotationModel, editPart.getModel());
+			if (marker != null) {
+				return marker;
 			}
+
+			annotationModel.reload();
```

Comment:
I believe the main problem with this approach is that this may bring back annotations for stale markers that have already been (manually) fixed in the editor compared to the saved state. If the quickfix is canceled, for example, these stale marker annotations may then remain visible in the editor. It also does not allow to use quickfixes for new problems only present in the editor, which consequently do not have a marker yet.

I still believe the only way to handle this situation properly with the present infrastructure is to create temporary markers for the quickfix.

Reply:
Correct me if I'm wrong, but as long as we don't save the editor or manually reload the Annotation model, it is not possible to get all Markers of the current state, right?
Or do you mean we create temporary markers for the whole editor using its typeEntry? In which case we would not use the Annotation Model at all.

Reply:
> Correct me if I'm wrong, but as long as we don't save the editor or manually reload the Annotation model, it is not possible to get all Markers of the current state, right?

The only way to get up-to-date markers for the current state in the editor would be to save and then wait for the builder to complete its run. A reload of the annotation model would only bring back the stale markers from the last save.

> Or do you mean we create temporary markers for the whole editor using its typeEntry? In which case we would not use the Annotation Model at all.

My initial idea was to create temporary markers based on the annotation model for the selected elements in the editor. However, this would limit quickfixes to the current selection when performed from inside the editor.

After thinking about this some more, maybe it also does not really make sense to perform bulk quickfixes from inside the editor in the first place. The Java and Xtext editors also do not seem support this.

Maybe we should instead focus on offering a simpler workflow in the editor, where users can select individual fixes quickly from the context menu. This would leave the bulk quickfixes, which are more powerful but also more slow due to the extra dialog, to the *Problems* view.

Reply:
I've changed it now, so that only the markers of the selected editparts are used. Maybe we can find a better solution in the future.
@mx990 I'm not sure if the creation/deletion of the temporary markers is done correctly.